packages/ui/src/components/Molecules/Pagination.tsx

```diff
@@ -0,0 +1,81 @@
+import { Link, useLocation } from '@custom/schema';
+import {
+  ArrowLongLeftIcon,
+  ArrowLongRightIcon,
+} from '@heroicons/react/20/solid';
+import clsx from 'clsx';
+import React from 'react';
+import { z } from 'zod';
+
+import { useIntl } from '../../utils/intl';
+
+export const paginationParamsSchema = z.object({
+  page: z.coerce.number().default(1),
+});
+
+export function useCurrentPage(): number {
+  const [location] = useLocation();
+  return paginationParamsSchema.parse(
+    Object.fromEntries(new URLSearchParams(location.search).entries()),
```

Comment:
I used `query-string` instead of `URLSearchParams`. Because it supports arrays by default.

Also, here is what I came up with for arrays in URL params:
```ts
const arrayOfStrings = z
  .array(z.coerce.string())
  .optional()
  .catch((ctx) => (typeof ctx.input === 'string' ? [ctx.input] : undefined));
```